include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,481 @@
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator>=(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right >= rhs.window_right) && (lhs.window_values_size >= rhs.window_values_size);
+    }
+    //!\}
+
+    //!\brief Pre-increment.
+    window_iterator & operator++() noexcept
+    {
+        get_minimiser();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    window_iterator operator++(int) noexcept
+    {
+        window_iterator tmp{*this};
+        get_minimiser();
+        return tmp;
+    }
+
+    //!\brief Return the minimiser.
+    value_type operator*() const noexcept
+    {
+        return minimiser_value;
+    }
+
+private:
+    //!brief Iterator to last element in range.
+    sentinel_t urange_end;
+
+    //!\brief The minimiser value.
+    size_t minimiser_value{0};
+
+    //!\brief Iterator to the rightmost value of one window.
+    it_t window_right;
+
+    //!\brief The number of values in one window.
+    uint32_t window_values_size;
+
+    //!\brief Stored values per window. It is necessary to store them, because a shift can remove the current minimiser.
+    std::deque<uint64_t> window_values;
+
+    //!\brief Increments iterator by 1.
+    void get_minimiser()
+    {
+        if (window_values.size() == 0)
+            window_first();
+        else // Call next_minimiser until minimiser value changed or end of the underlying range is reached.
+            while(!next_minimiser()){}
+    }
+
+    //!\brief Calculates minimisers for the first window.
+    void window_first()
+    {
+        for (uint32_t i = 0; (i < window_values_size - 1) ; i++)
+        {
+            window_values.push_back(*window_right);
+            std::ranges::advance(window_right,  1);
+        }
+        window_values.push_back(*window_right);
+        minimiser_value = *(std::min_element(std::begin(window_values), std::end(window_values)));
+    }
+
+    //!\brief Calculates the next minimiser value.
+    // For the following windows, we remove the first window value (is now not in window_values) and add the new
+    // value that results from the window shifting.
+    bool next_minimiser()
+    {
+        std::ranges::advance(window_right, 1);
+        if (window_right == urange_end)
+            return true;
+
+        uint64_t new_value = *window_right;
+        if (minimiser_value == *(std::begin(window_values)))
+        {
+            window_values.pop_front();
+            if (!window_values.empty())
+            {
+                window_values.push_back(new_value);
+                minimiser_value = *(std::min_element(std::begin(window_values), std::end(window_values)));
+                return true;
+            }
+
+        }
+        else
+        {
+            window_values.pop_front();
+        }
+
+        window_values.push_back(new_value);
```

Comment:
Since you `return` in the if-clause to jump out of the function and otherwise (else) continue will all of the code below, the else statement is unnecessary.
```suggestion

        window_values.pop_front();
        window_values.push_back(new_value);
```